python/test/unit/language/test_core_amd.py

```diff
@@ -913,46 +918,100 @@ def kernel(in_out_ptr):
+                         ((-1.0) ** sign) * (2.0 ** (exp - exp_bias)) * (1.0 + frac / (2.0 ** dtype.fp_mantissa_width))).float()
+
+    extended_exp = ((1 << (tl.float32.primitive_bitwidth - tl.float32.fp_mantissa_width - 1)) - 1) << tl.float32.fp_mantissa_width
+    # special cases, exp is 0b11..1
+    if dtype in [tl.float8e4, tl.float8e4b15]:
+        # float8e4m3 does not have infinities
+        output[fp == 0b01111111] = torch.nan
+        output[fp == 0b11111111] = torch.nan
+    else:
+        output = torch.where(exp == (1 << exp_width) - 1,
+                             ((sign << (tl.float32.primitive_bitwidth - 1)) | extended_exp | (frac << (tl.float32.fp_mantissa_width - dtype.fp_mantissa_width))).view(torch.float32),
+                             output)
+    return output
+
+
+@pytest.mark.parametrize("in_dtype", [torch.float16, torch.bfloat16])
+def test_convert_float16_to_float32(in_dtype, device):
+    """Tests that check convert_float_to_float32 function"""
+    check_type_supported(in_dtype, device)
+
+    f16_input = torch.tensor(range(-int(2 ** (16 - 1)), int(2 ** (16 - 1))), dtype=torch.int16).view(in_dtype)
+    f32_output = convert_float_to_float32(f16_input)
+
+    nan = f16_input.isnan()
+    assert torch.all(f32_output[nan].isnan())
+    inf = f16_input.isinf()
+    assert torch.all(f32_output[inf].isinf())
+    other = torch.logical_not(torch.logical_or(nan, inf))
+    assert torch.all(f16_input[other] == f32_output[other])
+
+
+def serialize_fp8(np_data, in_dtype):
+    return np_data
+# def serialize_fp8(np_data, in_dtype):
+#     if in_dtype == tl.float8e4b15:
+#         # triton's f8e4b15 format is optimized for software emulation
+#         # as a result, each pack of 4xfp8 values:
+#         # s0b0s1b1s2b2s3b3 (for s, b sign and bits respectively)
+#         # is actually internally stored as
+#         # s0s2b0b2s1s3b1b3
+#         # we apply the conversion here
+#         f8x4 = np_data.view(np.uint32)
+#         s = [(f8x4 & (0x80000000 >> i)) << i for i in range(0, 32, 8)]
+#         b = [(f8x4 & (0x7f000000 >> i)) << i for i in range(0, 32, 8)]
+#         signs = (s[0] >> 0) | (s[1] >> 16) | (s[2] >> 1) | (s[3] >> 17)
+#         bits = (b[0] >> 1) | (b[1] >> 17) | (b[2] >> 8) | (b[3] >> 24)
+#         # tensor of triton fp8 data
+#         return (signs | bits).view(np.int8)
+#     else:
+#         return np_data
+
+# inverse of `serialize_fp8`
+
+
+def deserialize_fp8(np_data, in_dtype):
+    return np_data
+# def deserialize_fp8(np_data, in_dtype):
+#     if in_dtype == tl.float8e4b15:
+#         f8x4 = np_data.view(np.uint32)
+#         s = [(f8x4 & (0x80000000 >> i)) << i for i in [0, 16, 1, 17]]
+#         b = [(f8x4 & (0x7f000000 >> i)) << i for i in [1, 17, 8, 24]]
+#         signs = (s[0] >> 0) | (s[1] >> 8) | (s[2] >> 16) | (s[3] >> 24)
+#         bits = (b[0] >> 0) | (b[1] >> 8) | (b[2] >> 16) | (b[3] >> 24)
+#         return (signs | bits).view(np.int8)
+#     else:
+#         return np_data
```

Comment:
Do we need this commented code?

Reply:
It is not needed as of this PR as packed float8e4b15 has not been enabled. But when upstream switch to the packed float8e4b15, the commented code will be uncommented.